### PR TITLE
[INV-3528] 'Suggested' ActionCreators Converted Over

### PR DESCRIPTION
**Pull request by LocalNewsTV**
# Overview
> A Portion of the work being done for #3428

This PR includes the following proposed change(s):

1. Draftstate for Activity is now typed, finding errors 
    - Stubbed the keys that were flagged as non-existent from this change
2. Created type for SuggestedTreatmentId
3. Finished converting `Suggestions` type actions into Action Creators
4. Tidied up typing, reducing warnings.

## Unused Keys

> Following keys were unused, and deleted.
 
- ACTIVITY_GET_SUGGESTED_PERSONS_REQUEST_OFFLINE 
- ACTIVITY_GET_SUGGESTED_TREATMENT_IDS_REQUEST_OFFLINE

## Testing
- Created form, Results from Dev / Local matched on Linked treatment Ids
- Created form, If you empty the string containing your name, it still repopulates it with your name

### CODE REVIEW

File: app/src/state/reducers/activity.ts [hunk 1]
```diff
@@ -24,37 +22,42 @@ import GeoShapes from 'constants/geoShapes';
 import { CURRENT_MIGRATION_VERSION, MIGRATION_VERSION_KEY } from 'constants/offline_state_version';
 import GeoTracking from 'state/actions/geotracking/GeoTracking';
 import Activity from 'state/actions/activity/Activity';
+import SuggestedTreatmentId from 'interfaces/SuggestedTreatmentId';
 
 export interface ActivityState {
   [MIGRATION_VERSION_KEY]: number;
   activity: any;
+  activeActivity: string | null;
+  activityErrors: any[];
```
Comment (LocalNewsTV) [inline app/src/state/reducers/activity.ts:32]: These two keys were being added/modified in the reducer, but were not present in the interface

File: app/src/state/reducers/activity.ts [hunk 2]
```diff
@@ -79,9 +82,9 @@ const initialState: ActivityState = {
   uiSchema: undefined
 };
 
-function createActivityReducer(): (ActivityState, AnyAction) => ActivityState {
+function createActivityReducer(): (ActivityState: ActivityState, AnyAction) => ActivityState {
   return (state = initialState, action) => {
-    return createNextState(state, (draftState) => {
+    return createNextState(state, (draftState: ActivityState) => {
```
Comment (LocalNewsTV) [inline app/src/state/reducers/activity.ts]: Matching the type to `draftState` here will now flag missing/mismatched keys in the reducer.
Adding this declaration flagged the keys being added that were missing in the interface.

Comment (plasticviking) [reply app/src/state/reducers/activity.ts]: I think it should be `Draft<ActivityState>`, since we're using `immer`

Comment (LocalNewsTV) [reply app/src/state/reducers/activity.ts]: @plasticviking  This is now corrected

**Pull request merged**